### PR TITLE
Bypass inappropriate/offensive warning without interaction

### PR DESCRIPTION
**Pull request by ner00**
Adding `CONTENT_CHECK_REQUIRED` status.

### CODE REVIEW

**Comment by zerodytrash** [comment]
hi,

Thanks for your contribution!

The content warning can be confirmed with one click? If so, I would recommend removing the warning by editing the response directly. Applying this bypass could result in some bugs. For example, videos with the "allow embed" function disabled would not be playable.

Do you have an example video with content warning for me? Then I'll see how this is passed in the response.

Thanks

**Comment by ner00** [comment]
Hi.

You're right, it can be bypassed with a button click. Although I don't know if the transition can be as smooth as with your generic approach to the actual "unavoidable" restriction, but I assume that you're correct and under several circumstances it would lead to playback issues. 

Unfortunately one or two of the videos that I came came across yesterday, which were flagged as inappropriate/offensive at the time, are now flagged as age-restricted. I've tried to search some potentially offensive content but just got age restriction at best, I assume I caught the videos in a limbo where the algorithm couldn't outright decide to restrict or ban.

Anyway, if I do find one eventually, I could share the link but that could also be re-categorized by the time you get to it. Is there anything specific that you would advise to catch in the request/response and/or overall objects on the page?

Cheers.

**Comment by zerodytrash** [comment]
My first assumption was that the status code `CONTENT_CHECK_REQUIRED` has no direct effect on the `playerResponse` and only exists to display the prompt by the UI. My idea was to simply set the `playabilityStatus` to `OK` to avoid the warning. But there I was wrong. The `playerResponse` is incomplete in such a case. After confirming the warning, the `playerResponse` is retrieved from youtube again with an extended URL-param named `bpctr`. This parameter contains a timestamp.

`/watch?v=<videoId>&pbj=1&bpctr=1619135880 `

tl;dr
Using the existing logic from this script really seems to be the easiest way to get around this warning (your solution). But since this script is only intended to bypass the unavoidable age restrictions, I would not implement it at the moment.

btw during my tests i used the following video, which currently shows this warning: https://www.youtube.com/watch?v=g5VOhBZIAYI

**Comment by ner00** [comment]
I assume the algorithm works differently depending on the fingerprint, the video you linked just gets me a standard `Age-restricted video (based on Community Guidelines)`, regardless if logged in or incognito.

In any case, it's fine, I'm just glad that I found your script and can tweak it to me needs, it's the best one yet from a few I've found when going around trying to deal with these restrictions. The offensive/inappropriate warning is just spam-y and redundant to me because logically it would necessarily overlap with age restriction (although YT shows either one or the other in the instances I encountered).
